src/ota/mod.rs

```diff
@@ -104,3 +126,2053 @@ impl Default for DeployStatus {
+            | OtaStatus::Rebooted
+            | OtaStatus::Success(_)
+            | OtaStatus::Error(_, _)
+            | OtaStatus::Failure(_, _) => false,
+        }
+    }
+
+    fn ota_id(&self) -> Option<OtaId> {
+        match self {
+            OtaStatus::Idle | OtaStatus::NoPendingOta | OtaStatus::Rebooted => None,
+            OtaStatus::Init(id)
+            | OtaStatus::Acknowledged(id)
+            | OtaStatus::Downloading(id, _)
+            | OtaStatus::Deploying(id, _)
+            | OtaStatus::Deployed(id)
+            | OtaStatus::Rebooting(id)
+            | OtaStatus::Success(id)
+            | OtaStatus::Error(_, id) => Some(id.clone()),
+            OtaStatus::Failure(_, id) => id.clone(),
+        }
+    }
+}
+
+impl Display for OtaStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            OtaStatus::Idle => write!(f, "Idle"),
+            OtaStatus::Init(req) => write!(f, "Init {req}"),
+            OtaStatus::NoPendingOta => write!(f, "NoPendingOta"),
+            OtaStatus::Acknowledged(req) => write!(f, "Acknowledged {req}"),
+            OtaStatus::Downloading(req, progress) => {
+                write!(f, "Downloading {req} progress {progress}")
+            }
+            OtaStatus::Deploying(req, progress) => write!(f, "Deploying {req} {progress}"),
+            OtaStatus::Deployed(req) => write!(f, "Deployed {req}"),
+            OtaStatus::Rebooting(req) => write!(f, "Rebooting {req}"),
+            OtaStatus::Rebooted => write!(f, "Rebooted"),
+            OtaStatus::Success(req) => write!(f, "Success {req}"),
+            OtaStatus::Error(err, req) => write!(f, "Error {req}: {err}"),
+            OtaStatus::Failure(err, req) => {
+                write!(f, "Failure")?;
+
+                if let Some(req) = req {
+                    write!(f, " {req}")?;
+                }
+
+                write!(f, ": {err}")
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct OtaId {
+    pub uuid: Uuid,
+    pub url: String,
+}
+
+impl From<OtaRequest> for OtaId {
+    fn from(value: OtaRequest) -> Self {
+        Self {
+            uuid: value.uuid.0,
+            url: value.url,
+        }
+    }
+}
+
+impl Display for OtaId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.uuid)
+    }
+}
+
+impl OtaStatus {
+    pub fn ota_request(&self) -> Option<&OtaId> {
+        match self {
+            OtaStatus::Acknowledged(ota_request)
+            | OtaStatus::Downloading(ota_request, _)
+            | OtaStatus::Deploying(ota_request, _)
+            | OtaStatus::Deployed(ota_request)
+            | OtaStatus::Rebooting(ota_request)
+            | OtaStatus::Success(ota_request)
+            | OtaStatus::Error(_, ota_request) => Some(ota_request),
+            OtaStatus::Failure(_, ota_request) => ota_request.as_ref(),
+            _ => None,
+        }
+    }
+}
```

Comment:
this is similar to `ota_id`. If not necessary you could remove it and use the othert